Sources/Harmony/Common/MemoryReporter.swift

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright 2018 Mobile Jazz SL
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import MachO
+
+
+/// A description of the usage of the RAM memory
+public struct RamState: CustomStringConvertible {
+    
+    let total: UInt64 // In bytes
+    let free: UInt64  // In bytes
+    let used: UInt64  // In bytes
+
+    public var description: String {
+        let usedInMB = used / 1024 / 1024
+        let freeInMB = free / 1024 / 1024
+        let totalInMB = total / 1024 / 1024
+
+        return "RAM: used: \(usedInMB)MB free: \(freeInMB)MB total: \(totalInMB)MB"
+    }
+}
+
+/// A utility that provides information about a system's memory
+public protocol MemoryReporter {
+    func ram() -> RamState?
```

Comment:
@newboadki if the struct is called "RamState" wouldn't make sense to call the function "ramState() -> RamState?" ? 

This is a side idea, if you think it is not meaningful, feel free to not apply any change here.

Reply:
Also, the method is not documented. Would be good to add documentation explaining what it means if it's nil and if it can be called from any thread.